driver/host_info.h

```diff
@@ -69,10 +71,18 @@ class HOST_INFO {
 private:
     const std::string HOST_PORT_SEPARATOR = ":";
     const std::string host;
+    const std::string host_id;
     const int port = NO_PORT;
 
     HOST_STATE host_state;
     bool is_writer;
 };
 
+inline std::ostream& operator<<(std::ostream& str, HOST_INFO v) {
+    char buf[1024];
+    sprintf(buf, "HostSpec[host=%s, port=%d, %s, %s]", v.get_host().c_str(), v.get_port(),
+            v.is_host_writer() ? "WRITER" : "READER", v.last_updated.c_str());
+    return str << std::string(buf);
```

Comment:
Can this be changed to the following? The `.c_str()`'s below may also be unnecessary.
```c++
return str << "HostSpec[host=" << v.get_host().c_str() << ", port=" << v.get_port() << ", " << v.is_host_writer() ? "WRITER" : "READER" << ", " << v.last_updated.c_str());
```

Reply:
Unfortunately `c_str()` is required